src/simulation/simulation/custom_gz_plugins/PathVisualizer.cc

```diff
@@ -0,0 +1,270 @@
+            max_angular_velocity = std::max(max_angular_velocity, std::sqrt(twist.angular().x() * twist.angular().x() + twist.angular().y() * twist.angular().y() + twist.angular().z() * twist.angular().z()));
+        }
+
+        if (max_linear_velocity == 0.0 && max_angular_velocity == 0.0) {
+            gzerr << "No path data to visualize" << std::endl;
+            return;
+        }
+
+        for (int i = 0; i < localPathData.poses_size(); ++i)
+        {
+            // Verify we have corresponding twist data
+            if (i >= localPathData.twists_size())
+                break;
+
+            auto pathMarker = this->scene->CreateCone();
+            if (!pathMarker)
+                continue;
+
+            auto pathMarkerVisual = this->scene->CreateVisual();
+            if (!pathMarkerVisual)
+                continue;
+
+            pathMarkerVisual->AddGeometry(pathMarker);
+            pathMarkerVisual->SetLocalScale(0.2, 0.2, 0.4);
+
+            // Set the position from the message
+            pathMarkerVisual->SetLocalPosition(
+                localPathData.poses(i).position().x(),
+                localPathData.poses(i).position().y(),
+                localPathData.poses(i).position().z());
+
+            // Use actual pose from the message
+            gz::math::Quaterniond yRotation(0.707, 0.0, 0.707, 0.0); // 90 degrees around y axis to align with the world frame
+            gz::math::Quaterniond originalRotation(
+                localPathData.poses(i).orientation().w(),
+                localPathData.poses(i).orientation().x(),
+                localPathData.poses(i).orientation().y(),
+                localPathData.poses(i).orientation().z());
+            // Combine rotations
+            gz::math::Quaterniond finalRotation = yRotation * originalRotation;
+            pathMarkerVisual->SetLocalRotation(finalRotation);
+
+            // Color based on velocity
+            auto material = this->scene->CreateMaterial();
+            if (!material)
+                continue;
+
+            auto twist = localPathData.twists(i);
+            auto linear_velocity = std::sqrt(
+                twist.linear().x() * twist.linear().x() +
+                twist.linear().y() * twist.linear().y() +
+                twist.linear().z() * twist.linear().z());
+            auto angular_velocity = std::sqrt(
+                twist.angular().x() * twist.angular().x() +
+                twist.angular().y() * twist.angular().y() +
+                twist.angular().z() * twist.angular().z());
+
+            auto linear_velocity_normalized = max_linear_velocity > 0.0 ? linear_velocity / max_linear_velocity : 0.0;
+            auto angular_velocity_normalized = max_angular_velocity > 0.0 ? angular_velocity / max_angular_velocity : 0.0;
+            auto red = 0.5 * linear_velocity_normalized + 0.5 * angular_velocity_normalized;
```

Comment:
Color is set based on the magnitude of both linear & angular velocities. We should decide at some point if this is optimal behavior.